### PR TITLE
snap: don't load unsupported implicit hooks.

### PR DESCRIPTION
**Pull request by kyrofa**
snapd currently filters out unsupported hooks when parsing the YAML, but not when loading them implicitly out of meta/hooks. This PR adds the filter on the implicit load as well.

### CODE REVIEW

File: snap/implicit.go [hunk 1]
```diff
@@ -139,6 +139,11 @@ func addImplicitHooksFromContainer(snapInfo *Info, snapf Container) error {
 }
 
 func addHookName(snapInfo *Info, hookName string) {
+	// Verify that the hook name is actually supported. If not, ignore it.
+	if !IsHookSupported(hookName) {
```
Comment (niemeyer) [inline snap/implicit.go]: It'd be slightly nicer to have this check done at the call site on line 135 instead. It's surprising an error prone to say doSomething and then just ignore it altogether. I'd expect an error, or a panic, but not complete silence and refusal to do what it's supposed to.

Comment (niemeyer) [inline snap/implicit.go]: (and line 115)

Comment (niemeyer) [inline snap/implicit.go]: Alternatively, rename the function to make the case more clear: `addHookIfValid` or something.

**Comment by niemeyer** [comment]
Tests also failed.

File: snap/implicit.go (hunk 1 above)
Comment (kyrofa) [inline snap/implicit.go]: Done. I went with renaming the function instead of duplicating code.

**Comment by kyrofa** [comment]
> Tests also failed.

Yeah, the merge with master seemed to make those happy.

**Pull request merged**